src/mavsdk/plugins/winch/winch_impl.cpp

```diff
@@ -0,0 +1,420 @@
+}
+
+void WinchImpl::process_status(const mavlink_message_t& message)
+{
+    mavlink_winch_status_t mavlink_winch_status;
+    mavlink_msg_winch_status_decode(&message, &mavlink_winch_status);
+
+    Winch::StatusFlags status_flags;
+    status_flags.healthy =
+        mavlink_winch_status.status & MAV_WINCH_STATUS_FLAG::MAV_WINCH_STATUS_HEALTHY;
+    status_flags.fully_retracted =
+        mavlink_winch_status.status & MAV_WINCH_STATUS_FLAG::MAV_WINCH_STATUS_FULLY_RETRACTED;
+    status_flags.moving =
+        mavlink_winch_status.status & MAV_WINCH_STATUS_FLAG::MAV_WINCH_STATUS_MOVING;
+    status_flags.clutch_engaged =
+        mavlink_winch_status.status & MAV_WINCH_STATUS_FLAG::MAV_WINCH_STATUS_CLUTCH_ENGAGED;
+    status_flags.locked =
+        mavlink_winch_status.status & MAV_WINCH_STATUS_FLAG::MAV_WINCH_STATUS_LOCKED;
+    status_flags.dropping =
+        mavlink_winch_status.status & MAV_WINCH_STATUS_FLAG::MAV_WINCH_STATUS_DROPPING;
+    status_flags.arresting =
+        mavlink_winch_status.status & MAV_WINCH_STATUS_FLAG::MAV_WINCH_STATUS_ARRESTING;
+    status_flags.ground_sense =
+        mavlink_winch_status.status & MAV_WINCH_STATUS_FLAG::MAV_WINCH_STATUS_GROUND_SENSE;
+    status_flags.retracting =
+        mavlink_winch_status.status & MAV_WINCH_STATUS_FLAG::MAV_WINCH_STATUS_RETRACTING;
+    status_flags.redeliver =
+        mavlink_winch_status.status & MAV_WINCH_STATUS_FLAG::MAV_WINCH_STATUS_REDELIVER;
+    status_flags.abandon_line =
+        mavlink_winch_status.status & MAV_WINCH_STATUS_FLAG::MAV_WINCH_STATUS_ABANDON_LINE;
+    status_flags.locking =
+        mavlink_winch_status.status & MAV_WINCH_STATUS_FLAG::MAV_WINCH_STATUS_LOCKING;
+    status_flags.load_line =
+        mavlink_winch_status.status & MAV_WINCH_STATUS_FLAG::MAV_WINCH_STATUS_LOAD_LINE;
+    status_flags.load_payload =
+        mavlink_winch_status.status & MAV_WINCH_STATUS_FLAG::MAV_WINCH_STATUS_LOAD_PAYLOAD;
+
+    Winch::Status new_status;
+    new_status.time_usec = mavlink_winch_status.time_usec;
+    new_status.line_length_m = mavlink_winch_status.line_length;
+    new_status.speed_m_s = mavlink_winch_status.speed;
+    new_status.tension_kg = mavlink_winch_status.tension;
+    new_status.voltage_v = mavlink_winch_status.voltage;
+    new_status.current_a = mavlink_winch_status.current;
+    new_status.temperature_c = mavlink_winch_status.temperature;
+    new_status.status_flags = status_flags;
+
+    set_status(new_status);
+
+    {
+        std::lock_guard<std::mutex> lock(_subscription_mutex);
+        _status_subscriptions.queue(
+            status(), [this](const auto& func) { _parent->call_user_callback(func); });
+    }
+}
+
+void WinchImpl::deinit() {}
+
+void WinchImpl::enable() {}
+
+void WinchImpl::disable() {}
+
+void WinchImpl::relax_async(uint32_t instance, const Winch::ResultCallback callback)
+{
+    MavlinkCommandSender::CommandLong command{};
+
+    command.command = MAV_CMD_DO_WINCH;
+    command.params.maybe_param1 = instance;
```

Comment:
Something like this maybe? For the -Werror in Windows

```suggestion
    command.params.maybe_param1 = static_cast<float>(instance);
```